File: event-bus-client/src/main/dotnet/Envelope.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace pegasus.eventbus.client
{
	public class Envelope
	{
		public byte[] Body { get; set; }
		public IDictionary<string, string> Headers { get; set; }
		
		
		public Envelope ()
		{
			this.Headers = new Dictionary<string, string>();
		}
		
		
		public virtual string GetHeader(string name)
		{
			string val = null;
			
			if (this.Headers.ContainsKey(name))
			{
				val = this.Headers[name];
			}
			
			return val;
		}
		
		public virtual void SetHeader(string name, string value)
		{
            if (false == this.Headers.ContainsKey(name))
            {
                this.Headers.Add(name, string.Empty);
            }

            this.Headers[name] = value;
		}

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("[REPLYTO=");
            sb.Append(this.GetReplyTo());
            sb.Append(",EVENT_TYPE=");
            sb.Append(this.GetEventType());
            sb.Append(",TOPIC=");
            sb.Append(this.GetTopic());
            sb.Append(",ID=");
            sb.Append(this.GetId().ToString());
            sb.Append(",CORRELATION_ID=");
            sb.Append(this.GetCorrelationId().ToString());
            sb.Append("]");

            return sb.ToString();
        }
	}
}

File: event-bus/event-bus-client/src/main/java/gov/ment/eventbus/client/Envelope.java
package gov.ment.eventbus.client;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

/**
 * General data structure used to represent the raw event data received form the
 * AMQP transport.
 */
public class Envelope {

  private UUID id;
  private UUID correlationId;
  private String topic;
  private String eventType;
  private String replyTo;
  private Date timestamp;
  private byte[] body = {};
  private Map<String, String> headers = new HashMap<String, String>();

  public byte[] getBody() {
    return body;
  }

  public UUID getCorrelationId() {
    return correlationId;
  }

  public String getEventType() {
    return eventType;
  }

  public Map<String, String> getHeaders() {
    return headers;
  }

  public UUID getId() {
    return id;
  }

  public String getReplyTo() {
    return replyTo;
  }

  public Date getTimestamp() {
    return timestamp;
  }

  public String getTopic() {
    return topic;
  }

  public void setBody(byte[] body) {
    this.body = body == null ? new byte[0] : body;
  }

  public void setCorrelationId(UUID correlationId) {
    this.correlationId = correlationId;
  }

  public void setEventType(String eventType) {
    this.eventType = eventType;
  }

  protected void setHeaders(Map<String, String> headers) {
    this.headers = headers;
  }

  public void setId(UUID id) {
    this.id = id;
  }

  public void setReplyTo(String replyTo) {
    this.replyTo = replyTo;
  }

  public void setTimestamp(Date timestamp) {
    this.timestamp = timestamp;
  }

  public void setTopic(String topic) {
    this.topic = topic;
  }

  @Override
  public String toString() {
    String[][] fields =
            { { "REPLYTO", getReplyTo() }, { "EVENT_TYPE", getEventType() },
                { "TOPIC", getTopic() }, { "ID", "" + getId() },
                { "CORRELATION_ID", "" + getCorrelationId() }, };
    StringBuffer sb = new StringBuffer("Envelope[");
    String sep = "";
    for (String[] key : fields) {
      String name = key[0];
      String val = key[1];
      if (val != null) {
        sb.append(sep + name + "=" + val);
        sep = ",";
      }
    }
    sb.append("]");
    return sb.toString();
  }
}
